persistence/mongodb-record/src/main/scala/net/liftweb/mongodb/record/field/BsonRecordField.scala

```diff
@@ -27,59 +27,70 @@ import net.liftweb.record._
 import com.mongodb._
 import org.bson.Document
 
+import scala.reflect.Manifest
 import scala.xml._
 
 /** Field that contains an entire record represented as an inline object value. Inspired by JSONSubRecordField */
-class BsonRecordField[OwnerType <: BsonRecord[OwnerType], SubRecordType <: BsonRecord[SubRecordType]]
-                        (rec: OwnerType, valueMeta: BsonMetaRecord[SubRecordType])(implicit subRecordType: Manifest[SubRecordType])
-  extends Field[SubRecordType, OwnerType]
-  with MandatoryTypedField[SubRecordType]
-{
-  def this(rec: OwnerType, valueMeta: BsonMetaRecord[SubRecordType], value: SubRecordType)
-          (implicit subRecordType: Manifest[SubRecordType]) = {
-      this(rec, value.meta)
-      set(value)
-  }
+abstract class BsonRecordTypedField[OwnerType <: BsonRecord[OwnerType], SubRecordType <: BsonRecord[SubRecordType]]
+(override val owner: OwnerType, valueMeta: BsonMetaRecord[SubRecordType])(implicit subRecordType: Manifest[SubRecordType])
+  extends Field[SubRecordType, OwnerType] {
 
   def this(rec: OwnerType, valueMeta: BsonMetaRecord[SubRecordType], value: Box[SubRecordType])
-          (implicit subRecordType: Manifest[SubRecordType]) = {
-      this(rec, valueMeta)
-      setBox(value)
+    (implicit subRecordType: Manifest[SubRecordType]) = {
+    this(rec, valueMeta)
+    setBox(value)
   }
 
-  def owner = rec
```

Comment:
It would still be publicly accessible, but I think your other comment about changing constructor param names also applies here. So, I'll rethink this.